analyzers/src/SonarAnalyzer.Common/Helpers/IsExternalInit.cs

```diff
@@ -18,8 +18,6 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
-#if NETFRAMEWORK
```

Comment:
Why do we need this change? This hack was supposed to be needed only for .Net Framework.

Reply:
The file was moved from UT project targeting 
* `net6.0` - works out of the box, #if made sense there
* `net48` - needs hack
to the `SonarAnalyzer.Common` project targeting
* net46 - needs hack
* netstandard2.0 - needs hack

So I had to remove the conditional compilation. It does no harm to the UT project.

Reply:
Very good point. This hack was previously part of the UTs where we could mock the missing part with low risk. 

On the other hand, I think it's not right to do this for production code since the analyzer will run also on PC with only .Net Framework where the syntax is not officially supported. We might have unexpected issues due to this.

I think it would be better to let the file in the UTs and find another approach to this problem.

Reply:
Based on [the docs](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.isexternalinit?view=net-6.0), this class is used only by a compiler to track metadata. It is not used during runtime and by-design should not change program behavior. There are also several blog posts suggesting describing this solution and a nugget package that adds this file.
I don't see this as a risk.

Reply:
If we want to go with this approach we should document the solution with at least a link to the blog posts and do proper testing on a machine that has only .Net Framework installed. An idea would be to do a manual build of the plugin, install it on a VM and try to analyze a reasonable big project to see if there are any errors or differences in the issues we raise.

Reply:
Only S2222 uses this `ProgramState` and it will not detect anything on any known project.
The init-only setter generates a classic setter in the MSIL form. So there is no change for CLR.
I'll add links to the class.

Reply:
I've added this link that describes the behavior as part of C# 9 specification:
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/init#metadata-encoding